builder/processor/src/main/java/io/helidon/builder/processor/CustomMethods.java

```diff
@@ -0,0 +1,284 @@
+    }
+
+    // static methods on prototype
+    private static GeneratedMethod factoryMethod(Errors.Collector errors,
+                                                 TypeContext.TypeInformation typeInformation,
+                                                 TypeName customMethodsType,
+                                                 List<String> annotations,
+                                                 Method customMethod) {
+
+        // if void: CustomMethodsType.methodName(param1, param2)
+        // if returns: return CustomMethodsType.methodName(param1, param2)
+        String generatedCall =
+                VOID_TYPE.equals(customMethod.returnType) ? "" : "return "
+                        + customMethodsType.genericTypeName().fqName() + "."
+                        + customMethod.name()
+                        + "("
+                        + customMethod.arguments().stream().map(Argument::name).collect(Collectors.joining(", "))
+                        + ")";
+
+        // factory methods are just copied to the generated prototype
+        return new GeneratedMethod(new Method(typeInformation.prototype(),
+                                              customMethod.name(),
+                                              customMethod.returnType(),
+                                              customMethod.arguments(),
+                                              customMethod.javadoc()),
+                                   annotations,
+                                   generatedCall);
+    }
+
+    private static List<CustomMethod> findMethods(TypeContext.TypeInformation typeInformation,
+                                                  TypeInfo customMethodsType,
+                                                  Errors.Collector errors,
+                                                  TypeName requiredAnnotation,
+                                                  MethodProcessor methodProcessor) {
+        // all custom methods must be static
+        // parameter and return type validation is to be done by method processor
+        return customMethodsType.elementInfo()
+                .stream()
+                .filter(TypeInfoPredicates::isMethod)
+                .filter(TypeInfoPredicates::isStatic)
+                .filter(TypeInfoPredicates.hasAnnotation(requiredAnnotation))
+                .map(it -> {
+                    // return type
+                    TypeName returnType = it.typeName();
+                    // method name
+                    String methodName = it.elementName();
+                    // parameters
+                    List<Argument> arguments = it.parameterArguments()
+                            .stream()
+                            .map(arg -> new Argument(arg.elementName(), arg.typeName()))
+                            .toList();
+
+                    // javadoc, if present
+                    List<String> javadoc = it.description()
```

Comment:
this should be delegated out to a helper for cases where the description is > 130 in length. I gave the same comment to David on ClassModel.

Reply:
line length is something that should be resolved when writing stuff out to file, not here.
The javadoc may be on a builder method (e.g. double nesting), and if you set formatting to use 4 spaces, you would already be 12 characters deep - so you would need to adjust this when printing it out.